app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/BaseException.java

```
@@ -36,4 +38,14 @@ public String getMessage() {
     public abstract String getDownstreamErrorCode();
 
     public abstract String getErrorType();
+
+    @SuppressWarnings("unchecked")
+    public <T extends BaseException> T hideStackTraceInLogs() {
+        hideStackTraceInLogs = true;
```

QQ: why is the default value to hide? Shouldn't it be false?